Introduce .net 5 for classic and preview deployment center.

nertim opened the pull request:
Please be warned there are some crazy hacks in here.. I have future WI (https://msazure.visualstudio.com/Antares/_workitems/edit/8695516) which will hopefully remove/remedy some of the hacks.

![image](https://user-images.githubusercontent.com/493476/97662562-abed2880-1a34-11eb-94df-fc76a488c597.png)

![image](https://user-images.githubusercontent.com/493476/97662584-b60f2700-1a34-11eb-8f97-8a20b1bdc911.png)


![image](https://user-images.githubusercontent.com/493476/97662648-d5a64f80-1a34-11eb-8599-b7fb0c3d7053.png)

![image](https://user-images.githubusercontent.com/493476/97662666-df2fb780-1a34-11eb-85c9-760ba2b3dfcb.png)

shimedh commented on server/src/stacks/2020-05-01/stacks/web-app-stacks/create/AspDotnet.ts:
```diff
@@ -5,6 +5,41 @@ export const aspDotnetCreateStack: WebAppCreateStack = {
   value: 'ASP.NET',
   sortOrder: 0,
   versions: [
```
Sorry just one question: Are we still using the 20202-05-01 stack? I thought we moved to the new one even for deployment center

nertim replied:
Oh in the classic deployment center we are still using the old API version. The new experience will go GA by end of the year so we decided to not spend the time updating the classic experience.

ammanifold commented:
Can you please add Screenshots?

ammanifold commented on client-react/src/pages/app/deployment-center/utility/GitHubActionUtility.ts:
```diff
@@ -154,7 +154,11 @@ export const getCodeAppWorkflowInformation = (
+      // NOTE(michinoy): In case of version 5, generate the dotnet core workflow file.
+      content =
+        runtimeVersion === '5' || runtimeVersion.toLocaleLowerCase() === 'v5.0'
+          ? getDotnetCoreGithubActionWorkflowDefinition(siteName, slotName, repoBranch, isLinuxApp, secretName, runtimeStackVersion)
```
Can you explain why we want dotnetcore here?

ammanifold replied:
NEvermind, sorry I missed the comment

ammanifold commented on client/src/app/site/deployment-center/deployment-center-setup/wizard-logic/deployment-center-state-manager.ts:
```diff
@@ -258,7 +260,9 @@ export class DeploymentCenterStateManager implements OnDestroy {
       }
     } else {
-      this.stack = runtimeStack;
+      // NOTE(michinoy): So it seems that in the stack API the stack value is 'asp.net', whereas from site config, the stack identifier is
```
stack value is actually aspnet

ammanifold commented on server/src/stacks/2020-05-01/stacks/web-app-stacks/create/AspDotnet.ts:
```diff
@@ -5,6 +5,41 @@ export const aspDotnetCreateStack: WebAppCreateStack = {
+          os: 'windows',
+          isPreview: false,
+          isDeprecated: false,
+          isHidden: false,
```
isHidden should be true

ammanifold commented on server/src/stacks/2020-05-01/stacks/web-app-stacks/create/AspDotnet.ts:
```diff
@@ -5,6 +5,41 @@ export const aspDotnetCreateStack: WebAppCreateStack = {
+          isPreview: false,
+          isDeprecated: false,
+          isHidden: false,
+          applicationInsightsEnabled: true,
```
applicationInsightsenabled should be false

ammanifold commented on server/src/stacks/2020-05-01/stacks/web-app-stacks/create/AspDotnet.ts:
```diff
@@ -5,6 +5,41 @@ export const aspDotnetCreateStack: WebAppCreateStack = {
+          os: 'linux',
+          isPreview: false,
+          isDeprecated: false,
+          isHidden: false,
```
same here, isHidden = true, appINishgtsEnabled = false

ammanifold commented on server/src/stacks/2020-05-01/stacks/web-app-stacks/create/AspDotnet.ts:
```diff
@@ -5,6 +5,41 @@ export const aspDotnetCreateStack: WebAppCreateStack = {
+          isHidden: false,
+          applicationInsightsEnabled: true,
+          remoteDebuggingEnabled: false,
+          runtimeVersion: 'v5.0',
```
For linux should be dotnetcore|5.0

ammanifold commented:
@nertim I assume we don't care about marking this as 'early access' in the deployment center?

ammanifold commented:
You will also need to update test cases: src/tests/Stacks/2020-05-01/web-app/validations.t

ammanifold commented:
Hey @nertim are you planning on hotfixing this in with .NET 5 hidden before release and then the only update on release date is to unhide stack? Or instead cherry pick this whole commit into release branch?

nertim commented:
@ammanifold Thank you for taking a look. 

I updated the values and test cases. No we do not need 'early access' in deployment center.

Now you brought up a very good point of how to enable this. I would rather have this all pushed in under a feature flag. Now this capability does not exist with the old API version, so I have added that.  Like this when it is time enable, the hotfix is simple setting the hidden property to true.

Can you please take a look again?

ammanifold commented:
> @ammanifold Thank you for taking a look.
> 
> I updated the values and test cases. No we do not need 'early access' in deployment center.
> 
> Now you brought up a very good point of how to enable this. I would rather have this all pushed in under a feature flag. Now this capability does not exist with the old API version, so I have added that. Like this when it is time enable, the hotfix is simple setting the hidden property to true.
> 
> Can you please take a look again?

Awesome, yes I will look through again! And thank you for adding the 'isHidden' capability, main reason I couldn't add this on 2020-05-01 with the stacks only pr. 

I think we should hotfix this PR ontop of what we have out currently. And then on .NET 5 release date it will just be a hotfix to the stacks to mark the these as unhidden (and change display text of ASP.NET to .NET)
